ListenerLeakTest.testListenerLeakOnMember_whenClientDestroyed

```
java.lang.AssertionError: expected:<0> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.client.listeners.leak.ListenerLeakTest.testListenerLeakOnMember_whenClientDestroyed(ListenerLeakTest.java:276)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/5578/testReport/junit/com.hazelcast.client.listeners.leak/ListenerLeakTest/testListenerLeakOnMember_whenClientDestroyed_smartRouting_false_/

```
Hiccups measured while running test 'testListenerLeakOnMember_whenClientDestroyed[smartRouting:false](com.hazelcast.client.listeners.leak.ListenerLeakTest):'
09:35:45, accumulated pauses: 1785 ms, max pause: 649 ms, pauses over 1000 ms: 0
09:35:50, accumulated pauses: 1980 ms, max pause: 638 ms, pauses over 1000 ms: 0
09:35:55, accumulated pauses: 224 ms, max pause: 27 ms, pauses over 1000 ms: 0
09:36:00, accumulated pauses: 351 ms, max pause: 232 ms, pauses over 1000 ms: 0
```

--- a/hazelcast-client/src/test/java/com/hazelcast/client/listeners/leak/ListenerLeakTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/listeners/leak/ListenerLeakTest.java
@@ -18,6 +18,7 @@ package com.hazelcast.client.listeners.leak;
 
 import com.hazelcast.client.config.ClientConfig;
 import com.hazelcast.client.impl.ClientEndpoint;
+import com.hazelcast.client.impl.ClientPartitionListenerService;
 import com.hazelcast.client.impl.clientside.ClientTestUtil;
 import com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl;
 import com.hazelcast.client.spi.impl.listener.AbstractClientListenerService;
@@ -257,14 +258,21 @@ public class ListenerLeakTest extends HazelcastTestSupport {
 
     @Test
     public void testListenerLeakOnMember_whenClientDestroyed() {
-        Collection<Node> nodes = createNodes();
+        final Collection<Node> nodes = createNodes();
 
         for (int i = 0; i < 100; i++) {
             newHazelcastClient().shutdown();
         }
 
-        for (Node node : nodes) {
-            assertEquals(0, node.getClientEngine().getPartitionListenerService().getPartitionListeningEndpoints().size());
-        }
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                for (Node node : nodes) {
+                    ClientPartitionListenerService listenerService = node.getClientEngine().getPartitionListenerService();
+                    assertEquals(0, listenerService.getPartitionListeningEndpoints().size());
+                }
+            }
+        });
     }
 }

